models/model_upload/multimodal_models/sglang-llama-3_2-11b-instruct/1/model.py

```diff
@@ -0,0 +1,240 @@
+    if self.quantization:
+      self.command += f" --quantization {self.quantization}"
+    if self.tensor_parallel_size:
+      self.command += f" --tensor-parallel-size {self.tensor_parallel_size}"
+    if self.mem_fraction_static:
+      self.command += f" --mem-fraction-static {self.mem_fraction_static}"
+    if self.context_length:
+      self.command += f" --context-length {self.context_length}"
+    if self.chat_template:
+      self.command += f" --chat-template {self.chat_template}"
+    if self.host:
+      self.command += f" --host {self.host}"
+    if self.port:
+      self.command += f" --port {self.port}"
+
+    try:
+      self.process = execute_shell_command(self.command)
+      wait_for_server(f'http://localhost:{self.port}')
+    except Exception as e:
+      if self.process:
+        logger.error("Terminating the sglang server process.")
+        terminate_process(self.process)
+      raise RuntimeError("Failed to start sglang server: " + str(e))
+
+
+def stream_completion(model, client, input_data, inference_params):
+  """Stream iteratively generates completions for the input data."""
+
+  temperature = inference_params.get("temperature", 0.7)
+  max_tokens = inference_params.get("max_tokens", 512)
+  top_p = inference_params.get("top_p", 1.0)
+
+  messages = construct_messages(input_data)
+  kwargs = dict(
+      model=model,
+      messages=messages,
+      temperature=temperature,
+      max_tokens=max_tokens,
+      top_p=top_p,
+      extra_body={"stop_token_ids": [151645, 151643]},
+      stream=True,
+  )
+  stream = client.chat.completions.create(**kwargs)
+
+  return stream
+
+
+class MyModel(ModelClass):
+  """A custom model that loads the model and generates text using SGLang Inference.
+  """
+
+  def load_model(self):
+    """Load the model here and start the openai sglang server."""
+
+    # SGLang parameters
+    self.mem_fraction_static = 0.9
+    self.tensor_parallel_size = 1
+    self.dtype = "half"
+    self.context_length = 4096
+    self.quantization = None
+    self.chat_template = "llama_3_vision"
+    self.host = "localhost"
+    self.port = 8761
+
+    self.server_manager = SGLangServerManager(self.mem_fraction_static, self.tensor_parallel_size,
+                                              self.dtype, self.context_length, self.quantization,
+                                              self.chat_template, self.port, self.host)
+
+    # Initialize the OpenAI client
+    openai_api_base = f"http://localhost:{self.port}/v1"
+    openai_api_key = "Not Required"
+    self.client = OpenAI(
+        api_key=openai_api_key,
+        base_url=openai_api_base,
+    )
+
+    python_executable = sys.executable
+
+    # if checkpoints section is in config.yaml file then checkpoints will be downloaded at this path during model upload time.
+    # checkpoints = os.path.join(os.path.dirname(__file__), "checkpoints")
+
+    checkpoints = "unsloth/Llama-3.2-11B-Vision-Instruct"
```

Comment:
maybe we want to do this really high up in the code as a variable so it serves better as an example